main/modules/importer/scrivener/importer.js

```diff
@@ -0,0 +1,764 @@
+                    id: cardId,
+                    title: cardTitle || i18n('Scene {num}', { num: beatId + 1 }),
+                    positionWithinLine: nextPositionInLine(
+                      jsonWithPlotlineSubfolderCards.cards,
+                      lineId,
+                      beatId
+                    ),
+                    description: descriptionArr,
+                    positionInBeat: beatId,
+                    beatId,
+                    lineId: childId,
+                    bookId,
+                  }),
+                }
+              },
+              newJsonWithBeats
+            )
+          }
+          return newJsonWithBeats
+        },
+        acc
+      )
+
+      const beatChildrenWithoutStoryLines = chaptersArray.filter((child) => {
+        return child['_attributes']['Type'] == 'Text' && child['Title']['_text']
+      })
+
+      const beatChildrenWithoutStoryLinesArr = Array.isArray(beatChildrenWithoutStoryLines)
+        ? beatChildrenWithoutStoryLines
+        : [beatChildrenWithoutStoryLines]
+
+      return beatChildrenWithoutStoryLinesArr.reduce((newJson, child, key) => {
+        const cardId = child['_attributes']['UUID'] || child['_attributes']['ID']
+        const cardTitle = child['Title']['_text']
+
+        // files same filename prefix with the cards (v2.7)
+        // same filename parent directory (v3)
+        const matchFiles = getMatchedRelevantFiles(relevantFiles, cardId)
+        let fileContents = { txtContent: createSlateParagraph(''), rtfContents: { lineId: 1 } }
+        if (matchFiles && matchFiles.length) {
+          const mappedFiles = mapMatchedFiles(newJson, matchFiles, fileContents, bookId)
+          fileContents = {
+            rtfContents: mappedFiles.rtfContents,
+            txtContent: createSlateEditor(mappedFiles.txtContent),
+          }
+        } else if (!newJson.lines.length) {
+          // if no plotline from rtf
+          createNewLine(newJson.lines, { position: 0, title: i18n('Main Plot') }, bookId)
+        }
+
+        const descriptionArr = Array.isArray(fileContents.txtContent)
+          ? fileContents.txtContent
+          : [fileContents.txtContent]
+
+        return {
+          ...newJson,
+          cards: createNewCard(newJson.cards, {
+            id: cardId,
+            title: cardTitle || i18n('Scene {num}', { num: beatId + 1 }),
+            positionWithinLine: nextPositionInLine(newJson.cards, lineId, beatId),
+            description: descriptionArr,
+            positionInBeat: beatId,
+            beatId,
+            lineId: fileContents.rtfContents.lineId,
+            bookId,
+          }),
+        }
+      }, withChildStoryLinesAdded)
+    }, withNewBeats)
+
+  return withNewCardsAndNewBeats
+}
+// Object, Array <any>, Object, Int -> Object { txtContent: {}, rtfContent: {} }
+function mapMatchedFiles(newJson, matchFiles, fileContents, bookId, isSection) {
+  matchFiles.flatMap((file) => {
```

Comment:
Similar problem here.  We discard the value, but we're using `flatMap`.
`flatMap` should be reserved for instances where we:
 - are interested in the result of `flatMap`ping, and
 - the function that we pass to `flatMap` produces an array itself.  (i.e. if we used ordinary `map` it would produce the type `[[Any]]`, but instead it'll create a "flattened" array `[Any]`.
 
 I think we should use `forEach` here again.